### PR TITLE
feat(leadspace): add support for background-media usage

### LINKED ISSUE #7148: [Lead space] Web Components:  Refactor to include Background media with Video & play-pause controls - DDS Consulting
###  Engineering info:
- HC engineer:  @andy-blum 
- HC Jira Ticket: https://jsw.ibm.com/browse/HC-2288
- DDS consulting engineer:   Ignacio

----

#### User Story

> As a `[user role below]`:
Carbon for ibm.com developer

> I need to:
change the `Lead space`

> so that I can:
provide the ibm.com adopter developers components they can use to build ibm.com web pages

#### Additional information

- Story within Storybook with corresponding knobs
- Utilize Carbon
- Create with Shadow DOM and Custom Elements standards
- **See the Epic (#6054) for the Design and Functional specs information**
- Includes only kaltura video support.  
- [Cloud v19  Kaltura video](https://pages.github.ibm.com/MSC-Cloud/hc_fractal/components/detail/5e--kaltura-video.html)
- Include as a Knob in Lead space: medium, tall and super
- Web Components Visual QA testing issue (#7152)
- Web Components Prod QA testing issue (#7154)
- [Lead space left aligned visual specs](https://ibm.ent.box.com/folder/94761007991?s=d9jey3rlfxqmxu059f0x9dtcx7f5ehuh). 
- [Lead space functional specs](https://ibm.ent.box.com/folder/94761007991?s=d9jey3rlfxqmxu059f0x9dtcx7f5ehuh). 
- [Background media visual specs](https://ibm.ent.box.com/folder/137924469929?s=t2y4gne6754xuf3popy9vfhr9rhupm0v). 
- [Background media functional specs](https://ibm.ent.box.com/folder/137924469929?s=t2y4gne6754xuf3popy9vfhr9rhupm0v).

#### Acceptance criteria
- [ ] Include README for the web component and corresponding styles
- [ ] Create Web Components styles in styles package
- [ ] Do not create knobs in Storybook that include JSON objects
- [ ] Break out Storybook stories into multiple variation stories, if applicable
- [ ] Create codesandbox example under `/packages/web-components/examples/codesandbox` and include in README
- [ ] Minimum 80% unit test coverage
- [ ] Update the Carbon for ibm.com website component [data file](https://github.com/carbon-design-system/carbon-for-ibm-dotcom-website/blob/master/src/data/components.json) to be sure the web site Component Status and Storybook links are correct
- [ ] Use the [Visual QA checklist](https://github.com/carbon-design-system/carbon-for-ibm-dotcom/wiki/Definition-of-done-(Visual-QA-checklist)) to verify design quality
- [ ] If a design is provided, the Designer is included as a Reviewer in the Pull Request
- [ ] Provide a direct link to the deploy preview for the designer in the Pull Request description
- [ ] A comment is posted in the Prod QA issue, tagging Praveen when development is finished

### PR DESCRIPTION
**Pull request by andy-blum**
### Related Ticket(s)

closes #7148 

### Description

Updates `dds-leadspace` component logic and styles to utilize `dds-background-media` for background imagery and videos

### Changelog

**Changed**

- `dds-leadspace` web component can use `dds-background-media` for background imagery and video

### CODE REVIEW

**Comment by IgnacioBecerra** [comment]
@andy-blum Overall looks really nice! The only noticeable difference between the regular and the refactored stories is that the gradient seems more prevalent in the refactored versions, just double checking if this was intended.

This might be more of a design question, but wondering if we should also get rid of the suggested videos at the end (Visit IBM Watson in the example)? They're not clickable, so they look a bit weird being in the leadspace. Should there be a better video that fit the leadspace aspect ratio properly?

Screen Shot 2021-12-01 at 2 14 48 PM

cc: @DragosRistici

**Comment by andy-blum** [comment]
@IgnacioBecerra I don't know of a way with the Kaltura API to hide the related videos popover. I wonder if that's something that's editable from the kaltura admin side when videos are added there.

**Comment by DragosRistici** [comment]
@andy-blum @IgnacioBecerra 

Hey, guys,

Besides the issue you emphasise above, I noticed that on these refactored stories the sm breakpoint does not follow the specs (it should be unchanged) and instead it is laid out like the medium refactored breakpoint. Maybe you devs used just one style for both medium and small up until now. 
Screenshot 2021-12-07 at 14 33 02

File: packages/web-components/src/components/leadspace/leadspace.scss
Comment (IgnacioBecerra) [inline packages/web-components/src/components/leadspace/leadspace.scss:39]:
```suggestion
  ::slotted(#{$dds-prefix}-background-media) {
    position: absolute;
    @include carbon--breakpoint(md) {
      inset: 0;
    }
  }
```

Following up on @DragosRistici's comment, modifying this rule like this should make the image display on the bottom as expected.

**Review by IgnacioBecerra — APPROVED** [review]
@andy-blum Looks fine to me, we just need a design approval. I'm assuming the old stories will be removed and the refactored version will become the new ones, so we just gotta make sure to clean that up before we merge this. Once that's done, all that'd be left would be to update snapshots :)

Thanks Andy!

**Comment by oliviaflory** [comment]
@andy-blum this is looking really good! 

Like Ignacio mentioned above we will want to remove "refactored" from the stories. Please use sentence case to name them we try to keep these consistent across all stories

Background media
- `Default`
- `With image`
- `With video`

Leadspace
- `Super`
- `Super with image`
- `Super with video`
- `Tall with image`
- `Tall with video`
- etc

---

There are a few things missing from the [functional specs](https://ibm.ent.box.com/file/877124759363), but we could open new issues to capture the following:
- **hotspot area** – the left 7 columns should also allow the viewer to click and play/pause the video. We don't need to address immediately because they can pause with the icon button in the corner
- **Lighbox media viewer/expressive modal** – this looks like another variation that opens a lightbox media viewer with a large centered play button on the left size. Ignacio mentioned this version may not have been presented initially and therefore not covered here.
- **Optional tool tip** – from the 5th item in the Functional specs, it looks like the Carbon tool tip is optional. I would have expected a toggle in the knobs to allow the user to turn it off/on. But the fact that it's there currently is more important.

**Comment by oliviaflory** [comment]
@andy-blum this looks great! Just two comments:

1. Percy is showing some changes on all Leadspace mobile breakpoints, are we swapping images for mobile or using the same one across all breakpoints when we use the `background media` component? 

Screen Shot 2022-01-19 at 1 33 46 PM


2. Just two stories need the capitalization fix:
- Medium with video
- Short with v
Screen Shot 2022-01-19 at 1 34 04 PM
ideo

**Comment by andy-blum** [comment]
@oliviaflory we're definitely still switching images based on device size, I think the issue here is the aspect ratio.

**Comment by andy-blum** [comment]
@oliviaflory fixes should be in place!

**Review by oliviaflory — APPROVED** [review]
Thank you @andy-blum LGTM!

**Review by IgnacioBecerra — APPROVED** [review]
LGTM, thanks for accommodating all the changes @andy-blum!

**Pull request merged**
